pkg/migrate/cstor/volume.go

```diff
@@ -0,0 +1,902 @@
+				List(metav1.ListOptions{
+					LabelSelector: "openebs.io/persistent-volume=" + v.PVName,
+				})
+			if err1 != nil {
+				return err1
+			}
+			if len(cvrList.Items) == 0 {
+				return errors.Errorf("no cvrs found for pv %s", v.PVName)
+			}
+			for _, replica := range cvrList.Items {
+				if replica.Status.Phase != cstor.CVRStatusOnline {
+					return errors.Errorf("failed to verify cvr %s phase expected: Healthy got: %s",
+						replica.Name, replica.Status.Phase)
+				}
+			}
+			return nil
+		})
+	if err != nil {
+		return err
+	}
+	err = retry.
+		Times(60).
+		Wait(5 * time.Second).
+		Try(func(attempt uint) error {
+			klog.Info("Waiting for cv to come to Healthy state")
+			cvObj, err1 := v.OpenebsClientset.CstorV1().
+				CStorVolumes(v.OpenebsNamespace).
+				Get(v.PVName, metav1.GetOptions{})
+			if err1 != nil {
+				return err1
+			}
+			if cvObj.Status.Phase != cstor.CStorVolumePhase("Healthy") {
+				return errors.Errorf("failed to verify cv %s phase expected: Healthy got: %s",
+					cvObj.Name, cvObj.Status.Phase)
+			}
+			return nil
+		})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (v *VolumeMigrator) patchTargetPodAffinity() error {
+	cvp, err := v.OpenebsClientset.CstorV1().
+		CStorVolumePolicies(v.OpenebsNamespace).
+		Get(v.PVName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	if cvp.Spec.Target.PodAffinity == nil {
+		return nil
+	}
+	klog.Info("Patching target pod with old pod affinity rules")
+	targetDeploy, err := v.KubeClientset.AppsV1().
+		Deployments(v.OpenebsNamespace).
+		Get(v.PVName+"-target", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	newTargetDeploy := targetDeploy.DeepCopy()
+	if newTargetDeploy.Spec.Template.Spec.Affinity == nil {
+		newTargetDeploy.Spec.Template.Spec.Affinity = &corev1.Affinity{}
+	}
+	newTargetDeploy.Spec.Template.Spec.Affinity.PodAffinity = cvp.Spec.Target.PodAffinity
+	data, err := GetPatchData(targetDeploy, newTargetDeploy)
+	if err != nil {
+		return err
+	}
+	_, err = v.KubeClientset.AppsV1().
+		Deployments(v.OpenebsNamespace).
+		Patch(v.PVName+"-target", types.StrategicMergePatchType, data)
+	return err
+}
+
+func (v *VolumeMigrator) cleanupOldResources() error {
+	klog.Info("Cleaning up old volume resources")
+	cvrList, err := cvr.NewKubeclient().
+		WithNamespace(v.OpenebsNamespace).
+		List(metav1.ListOptions{
+			LabelSelector: "openebs.io/persistent-volume=" + v.PVName,
+		})
+	for _, replica := range cvrList.Items {
+		rep := replica // pin it
+		rep.Finalizers = []string{}
+		_, err = cvr.NewKubeclient().
+			WithNamespace(v.OpenebsNamespace).
+			Update(&rep)
+		if err != nil {
+			return err
+		}
+		err = cvr.NewKubeclient().
+			WithNamespace(v.OpenebsNamespace).
+			Delete(replica.Name)
+		if err != nil {
+			return err
+		}
+	}
+	err = v.OpenebsClientset.CstorV1().
+		CStorVolumePolicies(v.OpenebsNamespace).
+		Delete(v.PVName, &metav1.DeleteOptions{})
+	if k8serrors.IsNotFound(err) {
+		return nil
```

Comment:
Don't return here. If pod restarts over here then the policy will not be removed from the CVC annotation and old CV also will not be cleaned.